### DISCUSSION TITLE
Ordering behavior parameters according to the order they were added

### DISCUSSION THREAD

**Discussion by Midhil457** [4ian/GDevelop#3502] [Enhancement ideas (and their technical discussions)]
It's really annoying that the behavior parameters are ordered alphabetically and not in the order they were added. It prevents makers from making planned and structured parameters (Ex: main toggle for a feature and then the parameters to customise that feature after that) and instead gets something that's hard to understand and looks confusing. 
Like this one:
![prar](https://user-images.githubusercontent.com/73597906/133931853-4f7672f1-2395-4e3b-8abc-ebe9b37bb8ba.png)

I don't see a use of sorting it alphabetically. Ordering it the other way makes more sense. Many users (or no one) will care if the parameters are alphabetically ordered or find a use from it. 

It could be a setting in the behavior settings to sort it alphabetically or not

So, a bonus would be to 'move up' and 'move down'  parameters in the behavior properties to easily structure/change parameter order afterwards. Currently reordering them means changing all the parameters. Also, the same could be helpful for the String options too
:)

**Comment by tristanbob** [comment]
I have also been frustrated by this, especially as my extensions get more and more parameters.  Important ones should be listed first.

**Comment by tristanbob** [comment]
Your bonus is also a really good idea.  When creating an extension, I may not have thought of all the parameters I need from the start.  When I am done, it is a lot of work to rearrange them because the sentence references don't change when you move parameters.  (This actually caused a bug on once with swapped parameters)

**Comment by 4ian** [comment]
@D8H made a partial solution with the ability to set properties into group. Going further we could imagine allowing to set things like the order in groups, ensure we order behavior parameters as they are created, and even collapse some groups or things like this.

**Comment by Elairyx7301** [comment]
What is the current status on this?
I believe this is is an important and highly useful change. I don't know why they are sorted by alphabet but this is tearing apart properties that belong together (e.g platformer character) like acceleration/deceleration, mixing up property orders so they are more confusing (listing specifics before the actual enable checkbox) or straight up positioning the lowest priority properties at the top and stealing all attention.
